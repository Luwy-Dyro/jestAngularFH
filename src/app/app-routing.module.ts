import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { ChariComponent } from './basic/chari/chari.component';
import { CounterComponent } from './basic/counter/counter.component';
import { CfatherComponent } from './basic/cfather/cfather.component';
import { CfatherSonComponent } from './basic/cfather-son/cfather-son.component';
import { CounterRouteComponent } from './basic/counter-route/counter-route.component';


const routes: Routes = [


    { path: 'basic/counter', component: CounterComponent  },
    { path: 'basic/counter/:initial', component: CounterRouteComponent  },
    { path: 'basic/chari', component: ChariComponent },
    { path: 'basic/cfather', component: CfatherComponent },
    { path: 'basic/cfather-son', component: CfatherSonComponent },
    { path: '**', redirectTo: 'basic/counter' },
    


];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
