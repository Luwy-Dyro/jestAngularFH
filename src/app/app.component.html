<!-- 
<h1>{{title}}</h1>
<hr>
<br> -->
<router-outlet></router-outlet>