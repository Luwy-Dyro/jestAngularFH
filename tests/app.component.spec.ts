import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '../src/app/app.component';

describe('AppComponent', () => {

  //Crea un modulo para las pruebas. 
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });




  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  it(`should have as title 'testeo01'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component.title).toEqual('testeo01');
  });


  // it('should render title of appComponent', () => {
  //   const fixture = TestBed.createComponent(AppComponent);
  //   fixture.detectChanges();
  //   const component = fixture.componentInstance;
  //   const compiled = fixture.nativeElement as HTMLElement;

  //   const h1 = compiled.querySelector('h1');
  //   //expect(compiled.querySelector('.content span')?.textContent).toContain('testeo01 app is running!');

  //   expect(h1?.textContent).toContain(component.title );
  // });

  // it('Debe hacer match con el snapchot', ()=>{

  //   const fixture = TestBed.createComponent(AppComponent);
  //   const compiled = fixture.nativeElement as HTMLElement;

  //   fixture.detectChanges();
  //   expect(compiled).toMatchSnapshot()

  // })

});
