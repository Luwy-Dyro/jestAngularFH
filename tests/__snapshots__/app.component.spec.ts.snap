// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Debe hacer match con el snapchot 1`] = `
<div
  id="root3"
>
  <router-outlet />
  
</div>
`;
